processor/processor.go

```diff
@@ -2232,26 +2282,65 @@ func (proc *HandleT) mainPipeline(ctx context.Context) {
 
 	chStore := make(chan storeMessage, bufferSize)
 	wg.Add(1)
+
 	go func() {
 		defer wg.Done()
 		defer close(chStore)
-
 		for msg := range chTrans {
 			chStore <- proc.transformations(msg)
 		}
 	}()
 
 	wg.Add(1)
 	go func() {
+		var mergedJob storeMessage
+		firstSubJob := true
 		defer wg.Done()
-		for msg := range chStore {
-			proc.Store(msg)
+		for subJob := range chStore {
+
+			if firstSubJob {
```

Comment:
to skip creating duplicate structs in case of unsplitted messages, you could add the following fragment above:

```
if firstSubJob && !subJob.haveMore {
	proc.Store(subJob)
	continue
}
```